test/e2e/vsphere_volume_vsan_policy.go

```diff
@@ -0,0 +1,281 @@
+		By("Invoking Test for VSAN policy hostFailuresToTolerate: 0, cacheReservation: 20")
+		scParameters[Policy_HostFailuresToTolerate] = "0"
+		scParameters[Policy_CacheReservation] = "20"
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		invokeValidVSANPolicyTest(client, namespace, scParameters)
+	})
+
+	// Valid policy.
+	It("verify VSAN storage capability diskStripes - 1 and objectSpaceReservation - 30 is honored for dynamically provisioned pvc using storageclass", func() {
+		By("Invoking Test for VSAN policy diskStripes: 1, objectSpaceReservation: 30")
+		scParameters[Policy_DiskStripes] = "1"
+		scParameters[Policy_ObjectSpaceReservation] = "30"
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		invokeValidVSANPolicyTest(client, namespace, scParameters)
+	})
+
+	// Valid policy.
+	It("verify VSAN storage capability diskStripes - 1 and objectSpaceReservation - 30 datastore - vsanDatastore is honored for dynamically provisioned pvc using storageclass", func() {
+		By("Invoking Test for VSAN policy diskStripes: 1, objectSpaceReservation: 30")
+		scParameters[Policy_DiskStripes] = "1"
+		scParameters[Policy_ObjectSpaceReservation] = "30"
+		scParameters[Datastore] = VsanDatastore
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		invokeValidVSANPolicyTest(client, namespace, scParameters)
+	})
+
+	// Valid policy.
+	It("verify VSAN storage capability objectSpaceReservation - 20 and iopsLimit - 100 is honored for dynamically provisioned pvc using storageclass", func() {
+		By("Invoking Test for VSAN policy objectSpaceReservation: 20, iopsLimit: 100")
+		scParameters[Policy_ObjectSpaceReservation] = "20"
+		scParameters[Policy_IopsLimit] = "100"
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		invokeValidVSANPolicyTest(client, namespace, scParameters)
+	})
+
+	// Invalid VSAN storage capabilties parameters.
+	It("verify VSAN storage capability objectSpaceReserve - 20 and stripeWidth - 2 is not honored for dynamically provisioned pvc using storageclass", func() {
+		By("Invoking Test for VSAN policy objectSpaceReserve: 20, stripeWidth: 2")
+		scParameters["objectSpaceReserve"] = "20"
+		scParameters[Policy_DiskStripes] = "2"
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		err := invokeInvalidVSANPolicyTestNeg(client, namespace, scParameters)
+		framework.Logf("Yo: %+q", err.Error())
+		errorMsg := "invalid option \\\"objectSpaceReserve\\\" for diskStripes in volume plugin kubernetes.io/vsphere-volume."
+		if !strings.Contains(err.Error(), errorMsg) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	// Invalid policy on a VSAN test bed.
+	// diskStripes value has to be between 1 and 12.
+	It("verify VSAN storage capability diskStripes - 14 and cacheReservation - 20 is not honored for dynamically provisioned pvc using storageclass", func() {
```

Comment:
maybe 14 and 20 should be `const` s? That way the `It`, the `By` text and the assignments will stay consistent. Also, not sure why you need 14 and 20 in both the `It` and `By`.